packages/react-router-dom/index.tsx

```diff
@@ -349,6 +349,20 @@ if (__DEV__) {
 
 export { ViewTransitionContext as UNSAFE_ViewTransitionContext };
 
+// TODO: (v7) Change the useFetcher data from `any` to `unknown`
+type FetchersContextObject = {
+  fetcherData: Map<string, any>;
+  register: (key: string) => void;
+  unregister: (key: string) => void;
+};
```

Comment:
Fetcher data lives in the React layer now, and we use ref counting via `register`/`unregister` to know when to remove from `fetcherData`